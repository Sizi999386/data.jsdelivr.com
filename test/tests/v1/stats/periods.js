const { expect } = require('chai');
const { makeEndpointAssertions, setupSnapshots } = require('../../../utils');

describe('/v1/stats/periods', () => {
	before(() => {
		setupSnapshots(__filename);
	});

	makeStatsPeriodsTests();
});

function makeStatsPeriodsTests () {
	makeEndpointAssertions('/v1/stats/periods', {}, [
		{
			params: {},
			assert: (response) => {
				expect(_.every(response.body, item => item.links.browsers)).to.be.true;
				expect(_.every(response.body, item => item.links.platforms)).to.be.true;
				expect(response.body).to.have.lengthOf(7);

				// Delete the floating periods which would mess up snapshots.
				response.body.splice(0, 1);
				expect(response).to.matchSnapshot();
			},
		},
	]);
}
