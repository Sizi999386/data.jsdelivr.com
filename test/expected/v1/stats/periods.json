{
	"/v1/stats/periods": [
		{
			"period": "2022-05",
			"periodType": "s-month",
			"links": {
				"browsers": "http://localhost:4454/v1/stats/browsers?period=2022-05",
				"network": "http://localhost:4454/v1/stats/network?period=2022-05",
				"packages": "http://localhost:4454/v1/stats/packages?period=2022-05",
				"platforms": "http://localhost:4454/v1/stats/platforms?period=2022-05",
				"proxies": "http://localhost:4454/v1/stats/proxies/{name}?period=2022-05"
			}
		},
		{
			"period": "2022-04",
			"periodType": "s-month",
			"links": {
				"network": "http://localhost:4454/v1/stats/network?period=2022-04",
				"packages": "http://localhost:4454/v1/stats/packages?period=2022-04",
				"proxies": "http://localhost:4454/v1/stats/proxies/{name}?period=2022-04"
			}
		},
		{
			"period": "2021",
			"periodType": "s-year",
			"links": {
				"browsers": "http://localhost:4454/v1/stats/browsers?period=2021",
				"platforms": "http://localhost:4454/v1/stats/platforms?period=2021"
			}
		},
		{
			"period": "2021-01",
			"periodType": "s-month",
			"links": {
				"browsers": "http://localhost:4454/v1/stats/browsers?period=2021-01",
				"platforms": "http://localhost:4454/v1/stats/platforms?period=2021-01"
			}
		},
		{
			"period": "2020",
			"periodType": "s-year",
			"links": {
				"browsers": "http://localhost:4454/v1/stats/browsers?period=2020",
				"platforms": "http://localhost:4454/v1/stats/platforms?period=2020"
			}
		},
		{
			"period": "2020-04",
			"periodType": "s-month",
			"links": {
				"browsers": "http://localhost:4454/v1/stats/browsers?period=2020-04",
				"platforms": "http://localhost:4454/v1/stats/platforms?period=2020-04"
			}
		},
		{
			"period": "2020-03",
			"periodType": "s-month",
			"links": {
				"browsers": "http://localhost:4454/v1/stats/browsers?period=2020-03",
				"platforms": "http://localhost:4454/v1/stats/platforms?period=2020-03"
			}
		},
		{
			"period": "2020-02",
			"periodType": "s-month",
			"links": {
				"browsers": "http://localhost:4454/v1/stats/browsers?period=2020-02",
				"platforms": "http://localhost:4454/v1/stats/platforms?period=2020-02"
			}
		}
	]
}
